andrada12/assignment3/WordGrid.java

```diff
@@ -0,0 +1,106 @@
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.Scanner;
+
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+
+/**
+ *
+ * @author andrada
+ */
+public class WordGrid {
+    int[] Ox = new int[]{-1,0,1};
+    int[] Oy = new int[]{-1,0,1};
+    
+    static class Position implements Comparable<Position>{
+        int x, y;
+        
+        public Position(int x, int y){
+            this.x = x;
+            this.y = y;
+        }
+        
+        public String toString(){
+            return x + " "+y;
+        }
+
+        @Override
+        public int compareTo(Position o) {
+            if (x == o.x && y == o.y)
+                return 0;
+            if (x == o.x)
+                return y - o.y;
+            return x - o.x;
+        }
+    }
+    
+    /* Initiates dfs from all the graph nodes */
+    public ArrayList<String> words(char[][] chars, String[] dictionary){
+       Dictionary.setDictionary(dictionary); 
+       boolean[][] visited = new boolean[chars.length][chars[0].length];       
+       String result = "";
+       ArrayList<String> allWards = new ArrayList<String>(); 
+       for (int i = 0; i< chars.length; i++)
+           for (int j = 0; j < chars[0].length; j++){
+               result = "" + chars[i][j];
+               dfs(i,j,chars,result,visited, allWards);
+               visited = new boolean[chars.length][chars[0].length];
+           }
+       return allWards;
+    }
+    
+    /* Given a position (x,y) it generates all its neighbours from a matrix */
+    public ArrayList<Position> neighbors(int x, int y, char[][] chars, String result, boolean[][] visited){
```

Comment:
Extracting a position class is nice design! Can you make it more object oriented? A few ideas:

- maybe pass a Position into this method instead of x,y
- move the neighbours() method to the Position class (you'd have to look again about how you handle visited)